healthcheck/healthcheck.go

```diff
@@ -127,11 +130,36 @@ func checkBackend(currentBackend *BackendHealthCheck) {
 	}
 }
 
+func (backend *BackendHealthCheck) newRequest(serverURL *url.URL) *http.Request {
+	// make a copy of the url
+	u := &url.URL{}
+	*u = *serverURL
+
+	if backend.Options.Port != 0 {
+		u.Host = net.JoinHostPort(u.Hostname(), strconv.Itoa(backend.Options.Port))
+	}
+	u.Path = u.Path + backend.Path
+
+	req := &http.Request{
```

Comment:
It seems inefficient to me to start with a URL, convert it to a string, and then have [NewRequest](https://golang.org/src/net/http/request.go?s=24894:24966#L748) parse it back into a URL.  In this code path, it probably doesn't matter.  I can change to NewRequest.